File: src/main/java/com/oyosite/ticon/specutils/component/ScoreboardComponentEntrypoint.java
package com.oyosite.ticon.specutils.component;

import com.oyosite.ticon.specutils.*;
import org.ladysnake.cca.api.v3.component.*;
import org.ladysnake.cca.api.v3.scoreboard.*;

public class ScoreboardComponentEntrypoint implements ScoreboardComponentInitializer {

	public static final ComponentKey<StaticEnderInkStorageComponent> ENDER_FLASK =
			ComponentRegistry.getOrCreate(SpectralUtilities.id("ender_flask"), StaticEnderInkStorageComponent.class);

	@Override
	public void registerScoreboardComponentFactories(ScoreboardComponentFactoryRegistry registry) {
		registry.registerScoreboardComponent(ENDER_FLASK, StaticEnderInkStorageComponent::new);

	}
}

File: src/main/kotlin/com/oyosite/ticon/specutils/component/ScoreboardComponentEntrypoint.kt
package com.oyosite.ticon.specutils.component

import com.oyosite.ticon.specutils.SpectralUtilities.Companion.id
import net.minecraft.server.MinecraftServer
import net.minecraft.world.scores.Scoreboard
import org.ladysnake.cca.api.v3.component.ComponentKey
import org.ladysnake.cca.api.v3.component.ComponentRegistry
import org.ladysnake.cca.api.v3.scoreboard.ScoreboardComponentFactoryRegistry
import org.ladysnake.cca.api.v3.scoreboard.ScoreboardComponentFactoryV2
import org.ladysnake.cca.api.v3.scoreboard.ScoreboardComponentInitializer

class ScoreboardComponentEntrypoint : ScoreboardComponentInitializer {
    override fun registerScoreboardComponentFactories(registry: ScoreboardComponentFactoryRegistry) {
        registry.registerScoreboardComponent<StaticEnderInkStorageComponent?>(
            ENDER_FLASK,
            ScoreboardComponentFactoryV2 { scoreboard: Scoreboard?, server: MinecraftServer? ->
                StaticEnderInkStorageComponent(
                    scoreboard,
                    server
                )
            })
    }

    companion object {
        val ENDER_FLASK: ComponentKey<StaticEnderInkStorageComponent?> =
            ComponentRegistry.getOrCreate<StaticEnderInkStorageComponent?>(
                id("ender_flask"), StaticEnderInkStorageComponent::class.java
            )
    }
}